### PR TITLE
Fix nested translators that rely on child translator tags

### PR DESCRIPTION
**Pull request by adomasven**
Accidentally discovered a bug while testing connector translation functionality

### CODE REVIEW

File: test/tests/translateTest.js [hunk 1]
```diff
@@ -571,6 +572,49 @@ describe("Zotero.Translate", function() {
 			assert.equal(pdf.attachmentLinkMode, Zotero.Attachments.LINK_MODE_IMPORTED_URL);
 			checkTestTags(pdf, true);
 		});
+		
+		it('child translators should not convert tags to canonical form', function* () {
```
Comment (dstillman) [inline test/tests/translateTest.js]: I know it's not done everywhere above, but let's try to keep these as sentences like they're intended:

`it('should not convert tags to canonical form in child translators')`

File: chrome/content/zotero/xpcom/translation/translate.js [hunk 2]
```diff
@@ -169,13 +173,14 @@ Zotero.Translate.Sandbox = {
...
 						notes.splice(j--, 1);
-					} else if(typeof(note) == "object") {
-						// Canonicalize tags
-						if(note.tags !== undefined) note.tags = translate._cleanTags(note.tags);
-					} else {
+					} else if(typeof(note) != "object") {
 						// Convert to object
 						notes[j] = {"note":note.toString()}
 					}
+					// If we're not in a child translator canonicalize tags
```
Comment (dstillman) [inline chrome/content/zotero/xpcom/translation/translate.js]: Here and above, add comma: "If we're not in a child translator, canonicalize tags"

File: chrome/content/zotero/tools/testTranslators/testTranslators.js [hunk 3]
```diff
@@ -549,18 +549,27 @@ function haveTranslators(translators, type) {
...
+			if(translatorTestView.canRun) {
+				translatorTestViewsToRun[type].push(translatorTestView);
+			}
+		}
+		
+		translatorTestStats[type].update();
+		var ev = document.createEvent('HTMLEvents');
+		ev.initEvent('ZoteroHaveTranslators-'+type, true, true);
+		document.dispatchEvent(ev);	
+	});
```
Comment (dstillman) [inline chrome/content/zotero/tools/testTranslators/testTranslators.js]: This is separate, right?

Comment (adomasven) [inline chrome/content/zotero/tools/testTranslators/testTranslators.js]: Yeah, I'll put this on a different branch, actually

**Pull request merged**